### PR TITLE
httphelper: Add handler for request logging

### PR DESCRIPTION
**Pull request by jvatic**

### CODE REVIEW

**Comment by jvatic** [comment]
This currently exposes the same functionality as the martini logging middleware, and should be expanded to satisfy #755

**Comment by titanous** [comment]
This should use log15 to do key=value logging

File: pkg/httphelper/request_logger.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+	log "github.com/flynn/flynn/Godeps/_workspace/src/gopkg.in/inconshreveable/log15.v2"
+	"github.com/flynn/flynn/pkg/random"
+)
+
+func NewRequestLogger(ctx log.Ctx, handler http.Handler) http.Handler {
+	l := log.New(ctx)
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		start := time.Now()
+		logger := l.New(log.Ctx{"req_id": random.UUID()})
```
Comment (titanous) [inline pkg/httphelper/request_logger.go]: This should be set to the `X-Request-ID` header if it exists.

File: pkg/httphelper/request_logger.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+		logger.Info("request started", "method", req.Method, "path", req.URL.Path)
```
Comment (titanous) [inline pkg/httphelper/request_logger.go]: Might be useful to include the source IP/port here too

**Comment by jvatic** [comment]
@titanous comments addressed

**Comment by jvatic** [comment]
moved into #762

**Pull request closed without merging**